pkg/kube/cluster-init.sh

```diff
@@ -544,15 +582,55 @@ change_to_new_token() {
 
 # monitor function to check if the cluster mode has changed, either from single node to cluster
 # or from cluster to single node
+#
+# Return values:
+#   0 - No action needed or transition initiated successfully
+#
+# Operational Cases:
+#
+# 1. NOT INITIALIZED: Skip checks until /var/lib/all_components_initialized exists
+#
+# 2. CLUSTER-TO-SINGLE TRANSITION (enc_status=2, enc_status_file missing):
+#    - If not in cluster mode: no action
+#    - Otherwise: cleanup registration, mark for single-node conversion, REBOOT
+#
+# 3. SINGLE-TO-CLUSTER TRANSITION (enc_status=0, no edge-node-cluster-mode flag):
+#    - EdgeNodeClusterStatus valid AND node was in single mode
+#    - Wait loop until valid enc_status received
+#    - Mark node as cluster mode before config changes
+#    - If zks registration exists: uninstall cluster components (kubevirt, longhorn) first
+#    - Bootstrap node case: rotate k3s token to controller-provided token
+#    - Remove old multus config, reassign with cluster node IP
+#    - Remove node labels for reapplication
+#    - Create transition pipe/flag for k3s restart coordination
+#    - Terminate k3s process
+#    - Non-bootstrap node join case: remove TLS certs, mark debuguser for reinit
+#    - Provision cluster config (bootstrap or join mode)
+#    - If enc_status_file disappears during wait for joining cluster: revert back to single-node, REBOOT
+#    - Non-bootstrap: create transition tracking file with timestamp, if joining cluster fails repeatedly, may REBOOT
+#    - Bootstrap: wait for k3s to start
+#    - Signal k3s restart via pipe, cleanup transition flag
+#
+# 4. ALREADY IN DESIRED MODE: No action taken
+#
+# 5. POST-CONVERSION REGISTRATION: If base-k3s-mode flag exists, uninstall kubevirt, longhorn, apply registration
+#
+# REBOOT SCENARIOS:
+# - Cluster-to-single: Always reboots after cleanup
+# - Single-to-cluster: Only non-bootstrap nodes may reboot if join fails (see check_cluster_transition_done) repeatedly
+# - Interrupted transition for non-bootstrap nodes: Reboots to single-node if enc_status_file disappears
 check_cluster_config_change() {
 
     # only check the cluster change when it's fully initialized
     if [ ! -f /var/lib/all_components_initialized ]; then
         return 0
     fi
 
-    if [ ! -f "$enc_status_file" ]; then
-      #logmsg "EdgeNodeClusterStatus file not found"
+    get_enc_status
+    enc_status=$?
```

Comment:
The preferred shell way to do this is 
enc_status=get_enc_status()
instead of using $?

shellcheck at least used to complain about using the old $?

Reply:
ok